lib/takeaway.rb

```diff
@@ -0,0 +1,51 @@
+require_relative 'menu'
+require_relative 'twiliosms'
+
+class Takeaway
+  INITIAL_COST = 0.00
+
+  def initialize(menu_object = Menu.new)
+    @menu_object = menu_object
+    @choice = []
+  end
+
+  def menu
+    @menu_object.list
+  end
+
+  def add_item(name, cost)
+    @menu_object.add(name, cost)
+  end
+
+  def pick(item)
+    @choice << @menu_object.list.select do |menu_item|
+      menu_item.name == item && menu_item.available?
+    end[0]
+  end
+
+  def display_basket
+    @choice
+  end
+
+  def sub_total
+    @cost = INITIAL_COST
+    add_prices
+    "£#{@cost.round(2)}"
+  end
```

Comment:
It would be better if you didn't store the @cost as an instance variable, as it duplicates information that already exists in the choice array. Here you are only using it between this method and add_prices. It would be a lot better to calculate on the fly rather than store in the variable as a buffer:

```rb
def sub_total
    cost = add_prices
    "£#{cost.round(2)}"
end

def add_prices(initial_cost = INITIAL_COST)
    cost = initial_cost
    @choice.each do |item|
        cost += convert_price(item)
    end
    cost
end
```